runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -1724,7 +1724,7 @@ VMnonNullSrcWrtBarCardCheckEvaluator(
       TR::InstOpCode::Mnemonic opSubtract = TR::InstOpCode::getSubstractOpCode();
       TR::InstOpCode::Mnemonic opCmpLog = TR::InstOpCode::getCmpLogicalOpCode();
       bool disableSrcObjCheck = true; //cg->comp()->getOption(TR_DisableWrtBarSrcObjCheck);
-      bool constantHeapCase = ((!comp->compileRelocatableCode()) && isConstantHeapBase && isConstantHeapSize && shiftAmount == 0 && (!is64Bit || TR::Compiler->om.generateCompressedObjectHeaders()));
+      bool constantHeapCase = ((!(comp->compileRelocatableCode() || comp->isOutOfProcessCompilation())) && isConstantHeapBase && isConstantHeapSize && shiftAmount == 0 && (!is64Bit || TR::Compiler->om.generateCompressedObjectHeaders()));
```

Comment:
I have tried backing out this change and I observe that `jit_tr_0` fails 1/5 times. The failure is the JITServer crashing during compilation while compiling a scorching method. I'm not sure if this code is related to the problem, but it definitely seems to make it more reproducible. So I'm going to leave this change in for now. Will update once solution is found to the crash mentioned above.